File: js/easy/q7.js
/*
 * You are a space explorer who has landed on a mysterious planet.
 * The riddle to unlock the treasure chamber is that the name of the chamber must be a palindrome.
 * Your mission is to help the robot check if the string is a palindrome correctly.
 * 
 * But the robot has some errors in its palindrome checking logic! 
 */

function isPalindrome(str) {
    let left = 0;
    let right = str.length - 1;

    
    while (left > right) {
        if (str[left] != str[right]) { 
            return "Not a palindrome"
        }
        left++;
    }

    return true;  
}

let word = "RaCecar";
console.log(isPaIindrome(word));  

File: python/easy/q7.py
"""
# You are a space explorer on a mysterious planet, and the key to unlocking the treasure chamber
# is ensuring the name of the chamber is a palindrome. However, the robot is making some errors.
# Help the robot by fixing the palindrome checking logic!
"""
def is_palindrome(s):
    left = 0
    right = len(s) - 1


    while left > right:
        if s[left] != s[right]: 
            return False 
        left += 1
    return True  

word = "raceCar"
print(is_palindrome(word))  # Expected: True or "Palindrome"
